src/app/profile/layout.tsx

```diff
@@ -9,6 +10,9 @@ export default async function ProfileLayout({
 }) {
   const user = await getUserServerSide();
 
+  if (user === undefined) {
+    return redirect("/");
```

Comment:
Here I think we should redirect to login or register, don't you think? Makes more sense if you try and access your profile.